alf/utils/dist_utils.py

```diff
@@ -361,23 +359,84 @@ def estimated_entropy(dist,
         calculating gradient
     """
     sample_shape = (num_samples, )
-    single_action = dist.sample(sample_shape=sample_shape, seed=seed)
-    if single_action.dtype.is_floating and assume_reparametrization:
+    single_action = dist.sample(sample_shape=sample_shape)
+    if single_action.dtype.is_floating_point and assume_reparametrization:
         entropy = -dist.log_prob(single_action)
         if check_numerics:
-            entropy = tf.debugging.check_numerics(entropy, 'entropy')
-        entropy = tf.reduce_mean(entropy, axis=0)
+            assert torch.all(torch.isfinite(entropy))
+        entropy = entropy.mean(dim=0)
         entropy_for_gradient = entropy
     else:
-        entropy = -dist.log_prob(tf.stop_gradient(single_action))
+        entropy = -dist.log_prob(single_action.detach())
         if check_numerics:
-            entropy = tf.debugging.check_numerics(entropy, 'entropy')
-        entropy_for_gradient = -0.5 * tf.math.square(entropy)
-        entropy = tf.reduce_mean(entropy, axis=0)
-        entropy_for_gradient = tf.reduce_mean(entropy_for_gradient, axis=0)
+            assert torch.all(torch.isfinite(entropy))
+        entropy_for_gradient = -0.5 * entropy**2
```

Comment:
Use math_ops.square() instead because **2 is much slower pytorch/pytorch#2618

Reply:
no longer the case?